src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -49,118 +49,72 @@ export class KeyStrokeHandler {
         config: ConfigurationEntry
     ): Promise<void> {
         try {
-            const changedText = this.getChangedText(event, config.isAutomatedTriggerEnabled, editor)
-            if (changedText === '') {
-                return
-            }
-            const autoTriggerType = this.getAutoTriggerReason(changedText)
-            if (autoTriggerType === '') {
-                return
-            }
-            const triggerTtype = autoTriggerType as CodewhispererAutomatedTriggerType
-            this.invokeAutomatedTrigger(triggerTtype, editor, client, config)
-        } catch (error) {
-            getLogger().error('Automated Trigger Exception : ', error)
-            getLogger().verbose(`Automated Trigger Exception : ${error}`)
-        }
-    }
+            if (!config.isAutomatedTriggerEnabled) return
 
-    getAutoTriggerReason(changedText: string): string {
-        for (const val of CodeWhispererConstants.specialCharactersList) {
-            if (changedText.includes(val)) {
-                this.specialChar = val
-                if (val === CodeWhispererConstants.lineBreak) {
-                    return 'Enter'
-                } else {
-                    return 'SpecialCharacters'
+            // Skip when output channel gains focus and invoke
+            if (editor.document.languageId === 'Log') return
+
+            // Pause automated trigger when typed input matches recommendation prefix for inline suggestion
+            if (InlineCompletion.instance.isTypeaheadInProgress) return
+
+            // Time duration between 2 invocations should be greater than the threshold
+            // This threshold does not applies to Enter | SpecialCharacters type auto trigger.
+            const duration = Math.floor((performance.now() - RecommendationHandler.instance.lastInvocationTime) / 1000)
```

Comment:
will this apply to all auto-trigger type now? but I can also remove this in my PR, so not a blocker

Reply:
yep, you can remove this part in ur PR Zoe, I try to keep everything in previous version and just do refactor